controllers/devworkspace/solver/che_routing.go

```diff
@@ -296,7 +296,9 @@ func (c *CheRoutingSolver) getGatewayConfigsAndFillRoutingObjects(cheCluster *v2
 	configs := make([]corev1.ConfigMap, 0)
 
 	// first do routing from main che-gateway into workspace service
-	if masterConfig := provisionMasterRouting(cheCluster, routing, cmLabels); masterConfig != nil {
+	if masterConfig, err := provisionMasterRouting(cheCluster, routing, cmLabels); err != nil {
```

Comment:
Sorry for probably dumn question: who is master? Is it a term from traefik?

Reply:
not a dumb question. It is not coming from traefik. The routing with authorization is now done in 2 places:
 1. che-gateway in eclipse-che namespace
 2. traefik instance in workspace pod
This function is configuring 1.

Reply:
That's internals but still better to avoid master wording.
What could be alternatives? Like facade/internal, public/internal.
Ideally, it's described in the doc.go and their logic live in separate files, so it's easier to get an overall picture, but decoupling logic into files is always I ask for but it never related to any PR :-D So, feel free to ignore.

Reply:
I've renamed "master" into "main".

I got your refactoring ideas with organizing it into multiple files, but this PR already contains a lot of refactoring so I would not make it more complex. The more PRs is coming to this area, so we will try to improve the code iteratively with every changeset.